components/skus/browser/rs/cxx/src/lib.rs

```diff
@@ -0,0 +1,462 @@
+    pub  extern "C" fn(
+        callback_state: *mut ffi::RefreshOrderCallbackState,
+        result: ffi::SkusResult,
+        order: &str,
+    ),
+);
+
+unsafe impl ExternType for RefreshOrderCallback {
+    type Id = type_id!("skus::RefreshOrderCallback");
+    type Kind = cxx::kind::Trivial;
+}
+
+async fn refresh_order_task(
+    sdk: Rc<skus::sdk::SDK<NativeClient>>,
+    callback: RefreshOrderCallback,
+    callback_state: UniquePtr<ffi::RefreshOrderCallbackState>,
+    order_id: String,
+) {
+    match sdk
+        .refresh_order(&order_id)
+        .await
+        .and_then(|order| serde_json::to_string(&order).map_err(|e| e.into()))
+        .map_err(|e| e.into())
+    {
+        Ok(order) => callback.0(callback_state.into_raw(), ffi::SkusResult::Ok, &order),
+        Err(e) => callback.0(callback_state.into_raw(), e, ""),
+    }
+}
+
+#[repr(transparent)]
+pub struct FetchOrderCredentialsCallback(
+    pub  extern "C" fn(
+        callback_state: *mut ffi::FetchOrderCredentialsCallbackState,
+        result: ffi::SkusResult,
+    ),
+);
+
+unsafe impl ExternType for FetchOrderCredentialsCallback {
+    type Id = type_id!("skus::FetchOrderCredentialsCallback");
+    type Kind = cxx::kind::Trivial;
+}
+
+async fn fetch_order_credentials_task(
+    sdk: Rc<skus::sdk::SDK<NativeClient>>,
+    callback: FetchOrderCredentialsCallback,
+    callback_state: UniquePtr<ffi::FetchOrderCredentialsCallbackState>,
+    order_id: String,
+) {
+    match sdk
+        .fetch_order_credentials(&order_id)
+        .await
+        .map_err(|e| e.into())
+    {
+        Ok(_) => callback.0(callback_state.into_raw(), ffi::SkusResult::Ok),
+        Err(e) => callback.0(callback_state.into_raw(), e),
+    }
+}
+
+#[repr(transparent)]
+pub struct PrepareCredentialsPresentationCallback(
+    pub  extern "C" fn(
+        callback_state: *mut ffi::PrepareCredentialsPresentationCallbackState,
+        result: ffi::SkusResult,
+        presentation: &str,
+    ),
+);
+
+unsafe impl ExternType for PrepareCredentialsPresentationCallback {
+    type Id = type_id!("skus::PrepareCredentialsPresentationCallback");
+    type Kind = cxx::kind::Trivial;
+}
+
+async fn prepare_credentials_presentation_task(
+    sdk: Rc<skus::sdk::SDK<NativeClient>>,
+    callback: PrepareCredentialsPresentationCallback,
+    callback_state: UniquePtr<ffi::PrepareCredentialsPresentationCallbackState>,
+    domain: String,
+    path: String,
+) {
+    match sdk
+        .prepare_credentials_presentation(&domain, &path)
+        .await
+        .map_err(|e| e.into())
+    {
+        Ok(Some(presentation)) => callback.0(
+            callback_state.into_raw(),
+            ffi::SkusResult::Ok,
+            &presentation,
+        ),
+        Ok(None) => callback.0(callback_state.into_raw(), ffi::SkusResult::Ok, ""),
+        Err(e) => callback.0(callback_state.into_raw(), e, ""),
+    }
+}
+
+#[repr(transparent)]
+pub struct CredentialSummaryCallback(
+    pub  extern "C" fn(
```

Comment:
nit: double space

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/780676bbebb2283387763229a4592c4eb4eadb9d